source/common/router/retry_state_impl.cc

```diff
@@ -225,6 +225,10 @@ RetryStatus RetryStateImpl::shouldRetry(bool would_retry, DoRetryCallback callba
   }
 
   if (retries_remaining_ == 0) {
```

Comment:
Consider throwing in a comment here on what exactly this means (exhausting retry allotment configured in the route). The terminology in this part of the code is a bit treacherous.